### PR TITLE
WIP: Refactored my channel endpoints

### PR DESCRIPTION
**Pull request by devos50**
In this PR, I refactored the endpoints regarding your channel to not make any distinguishment between your own channel and the channel of other peers. This allows to edit any arbitrary channel (given that the channel state is `open` or `semi-open`).

I mostly moved files around. I've added some tests to check for the `unauthorized` error (401) when performing requests and I created a new endpoint to fetch general information about a specific discovered channel (`/channels/discovered/{channelcid}`.

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis can you review? I don't think it's necessary to check all 1000 lines since it mostly consists of moving code around. Please check the doc changes to get a better idea of what I've done exactly.

File: doc/Tribler REST API.md [hunk 1]
```diff
@@ -127,6 +127,36 @@ Returns all discovered channels in Tribler.
 }
 ```
 
+## `PUT /channels/discovered/{channelcid}`
```
Comment (lfdversluis) [inline doc/Tribler REST API.md]: This makes no sense to me. By going to the discovered channels with a channel id parameter I create my own channel? What if I use a channel id already in use by someone else? This endpoint is confusing imo.

Comment (devos50) [inline doc/Tribler REST API.md]: I'm sorry about that, that was a typo. Should be `PUT /channels/discovered`.

**Comment by lfdversluis** [comment]
Too many changes to pick what is merely moved. Git should see this as a rename if you copy paste code like with my creation of the tunnel helper twistd plugin?

File: Tribler/Core/Modules/restapi/channels/base_channels_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        request.setResponseCode(http.UNAUTHORIZED)
+        return json.dumps({"error": message})
+
+    def get_channel_from_db(self, cid):
+        """
+        Returns information about the channel from the database. Returns None if the channel with given cid
+        does not exist.
+        """
+        channels_list = self.channel_db_handler.getChannelsByCID([cid])
+        if not channels_list:
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/channels/base_channels_endpoint.py]: Wouldn't it be nicer to do `return None if not channels_list else channels_list[0]`?

**Comment by lfdversluis** [comment]
Gave feedback where I could, but 1k additions and 1.1k deletions of file movement is hard to filer :/

**Comment by devos50** [comment]
@whirm want me to split this PR into multiple ones?

**Comment by whirm** [comment]
Yes please, do it in manageable steps (at least at the commit level)

**Comment by whirm** [comment]
retest this please

**Comment by devos50** [comment]
@whirm you mean commit-wise or by creating multiple PRs?

**Comment by whirm** [comment]
You can keep a single PR if you slice it into small self-contained commits that can be independently reviewed

**Comment by devos50** [comment]
@whirm ok great, I will work on that!

**Comment by devos50** [comment]
I will create a new PR for this later, I've cherry-picked these changes and I will continue working in my own branch.

**Pull request closed without merging**